DOM.js

```diff
@@ -0,0 +1,154 @@
+//Dummy function to return API calls
+
+function lookUpPokeApi(input, callback) {
+
+  var pokeApiResponse = {
+    name: "bulbasaur",
+    entryNumber: 1,
+    moves: ["captivate", "razor wind", "swords dance"],
+    type: ["poison", "grass"],
+    description: "Bulbasaur can be seen napping in bright sunlight. There is a seed on its back. By soaking up the sun’s rays, the seed grows progressively larger.",
+    sprite: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
+  }
+
+  callback(pokeApiResponse);
+}
+
+// function to randomly assign new gif
+function lookUpGiphy(input, callback) {
+
+  var giphyResponse = [
+    "https://media.giphy.com/media/I2nZMy0sI0ySA/giphy.gif",
+    "https://media.giphy.com/media/SfjCRiDNA951m/giphy.gif",
+    "https://media.giphy.com/media/51S87GVazTD7W/giphy.gif",
+    "https://media.giphy.com/media/imoIuptU6lelW/giphy.gif"
+  ];
+
+  var randomGif = Math.floor(Math.random() * 3);
+
+  callback(giphyResponse[randomGif]);
+
+}
+
+
+//IIFE (Not Eevee...)
+(function(){
+
+  //define vars for search button and input
+  var searchButton = document.querySelector('#searchButton');
+  var searchInput = document.querySelector('input[name="searchInput"]');
+  var pokemonDetails = document.querySelector('#pokemon-details')
+  var pokemonArticle = document.querySelector('#pokemon');
+
+  //add event listener for button click
+  searchButton.addEventListener('click', function(){
+    lookUpPokeApi(searchInput.value, pokeCallback);
+    lookUpGiphy(searchInput.value, gifCallback);
+  });
+
+  //abstract function to add text elements within pokemon details
+  addNewNode = function(parentNodeId, element, text) {
```

Comment:
does this need to be declared with a var?

Reply:
Yes you very observant person :)